Project/VetAppointmentAPI/VetAppointmentAPI/VetAppointment.WebAPI/Controllers/DrugsController.cs

```diff
@@ -13,85 +19,67 @@ namespace VetAppointment.WebAPI.Controllers
     public class DrugsController : ControllerBase
     {
         private readonly IDrugRepository drugRepository;
-        private readonly IValidator<CreateDrugDto> drugValidator;
-        private readonly IMapper mapper;
-        public DrugsController(IDrugRepository drugRepository, IValidator<CreateDrugDto> validator, IMapper mapper)
-        {
-            this.drugRepository = drugRepository;
-            this.drugValidator = validator;
-            this.mapper = mapper;
-        }
+        private readonly IMediator mediator;
 
-        [HttpGet]
-        public async Task<IActionResult> SearchDrugs([FromQuery] string? title, [FromQuery] int? price)
+        public DrugsController(IDrugRepository drugRepository,IMediator mediator)
         {
-            var drugStocks = await drugRepository.Find(x => x.Title == title || x.Price == price);
-            if (drugStocks != null)
-                return Ok(drugStocks);
-            return NotFound();
+            this.drugRepository = drugRepository;
+            this.mediator = mediator;
         }
 
         [HttpGet("GetAll")]
-        public async Task<IActionResult> GetAllDrugs()
+        public async Task<List<DrugResponse>> GetAllDrugs()
         {
-            var drugs = await drugRepository.All();
-            return drugs != null ? Ok(drugs) : NotFound();
+            return await mediator.Send(new GetAllDrugsQuery());
```

Comment:
This one seems to use the Send method from the mediator. Maybe that needs to also be used for the others?